circuit/breaker.py

```diff
@@ -85,7 +99,10 @@ def test(self):
         """
         if self.state == 'open':
             delta = self.clock() - self.last_change
-            if delta < self.reset_timeout:
+            delay_time = self.backoff_cap and min(self.reset_timeout * (2 ** self.test_fail_count), self.backoff_cap) or self.reset_timeout
```

Comment:
This line is a bit hard to crack, might have to get a bit creative with intermediary variable. Something like:

```python
delay_time = self.reset_timeout
if self.backoff_cap:
    delay_time = self.reset_timeout * (2 ** self.test_fail_count)
    delay_time = min(delay_time, self.backoff_cap)
```